### PR TITLE
Update nwm-archive.yaml

### PR DESCRIPTION
**Pull request by Patrick-Keown**
Adding v2.1 information and Zarr buckets.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: datasets/nwm-archive.yaml [hunk 1]
```diff
@@ -43,6 +45,22 @@ Tags:
   - sustainability
 License: Open Data. There are no restrictions on the use of this data.
 Resources:
+  - Description: The complete archive of NWM input forcing and model output data version 2.1 in NetCDF format.
+    ARN: arn:aws:s3:::noaa-nwm-retrospective-2-1-pds
+    Region: us-east-1
+    Type: S3 Bucket
+    '[AWS CLI] (https://aws.amazon.com/cli/)' Access (No AWS account required):
+    aws s3 ls s3://noaa-nwm-retrospective-2-1-pds/ --no-sign-request
```
Comment (jflasher) [inline datasets/nwm-archive.yaml:53]: Thanks @Patrick-Keown. Looks like some extra stuff got added in here, `Type` should just be `S3 Bucket`. So if you remove lines 52-53 and again down at 60-61, I think this should pass.

**Review by jflasher — CHANGES_REQUESTED** [review]

**Comment by Patrick-Keown** [comment]
Fixed but not sure if it resubmitted


Patrick Keown

Program Manager, NOAA Big Data Program *(Temporary Detail)*

Office of the Chief Information Officer (OCIO)

Geospatial Program Manager

Office of Coast Survey (OCS)

National Oceanic & Atmospheric Administration

(615) 319-5906 | ***@***.***

"Be sure when you step, step with care and great tact" - Dr. Seuss

File: datasets/nwm-archive.yaml [hunk 2]
```diff
@@ -67,6 +81,13 @@ DataAtWork:
       URL: https://nbviewer.jupyter.org/gist/rsignell-usgs/d3dfaf3cd3d8b39894a69b22127dfe38
       AuthorName: Rich Signell
       AuthorURL: https://www.usgs.gov/staff-profiles/richard-p-signell
+    - Title: Explore Repository of Tutorials on National Water Model V2.1 Retrospective Dataset in Zarr
+      URL: https://github.com/NCAR/rechunk_retro_nwm_v21/tree/main/notebooks
+      AuthorName: James McCreight
+      AuthorURL: https://github.com/jmccreight
+  Zarr v2.1 Conversion Code Repository:
```
Comment (jflasher) [inline datasets/nwm-archive.yaml:88]: @Patrick-Keown sorry, didn't see this earlier. This block is problematic. It needs to be under either a `Publications`, `Tools & Applications`, or `Tutorials` heading like the others. And it'll need to have a URL as well. Please also be careful of the hypens, there should be just one for the whole item (like the ones above).

**Review by jflasher — CHANGES_REQUESTED** [review]

File: datasets/nwm-archive.yaml (hunk 2 above)
Comment (Patrick-Keown) [reply datasets/nwm-archive.yaml]: Thanks Joe. Issued resolved.

**Pull request merged**